programmers/난이도별/level01.약수의_개수와_덧셈/JiaLee0707.js

```diff
@@ -0,0 +1,15 @@
+function solution(left, right) {
+    let result = 0;
+    [...Array(right-left+1).keys()].map((value) => {
+        let count = 0;
+        const loopLenght = value + left;
+        for (let i = 1; i<=loopLenght; i++) {
+            if (loopLenght % i === 0) count += 1;
+        }
```

Comment:
꾸준하게 PR 올리시는 모습 정말 존경합니다 ^!^

같은 부분이 반복적으로 실행된다면 함수로 따로 빼는 것이 코드 **재 사용성**이 높아지고, **가독성**도 좋다고 합니다!
무엇보다, 함수를 추상화하는 과정에서 각 함수의 **책임이 명백**해진다고 하네요 👍

```javascript
function getCount(loopLength){
  let count = 0;

  for (let i = 1; i<=loopLenght; i++) {
     if (loopLenght % i === 0) count += 1;
  }

  return count;
}
```

```suggestion
        const loopLenght = value + left;
        let count = getCount(loopLength);
```

Reply:
요거 프로그래머스에서 확인해도 정답이 동일하게 나올까요?

Reply:
네!

~~중간에 변수명이 오타나서 황급히 수정했네요~~
코드 흐름상의 문제는 없었습니다_!

아래에가 [JiaLee0707](https://github.com/Mirim-Study/Algorithm/commits?author=JiaLee0707)님의 코드를 돌린 결과값이고, 위가 변경사항으로 돌려본 결과값입니다!

<img width="374" alt="스크린샷 2022-03-30 오전 9 09 45" src="https://user-images.githubusercontent.com/48716775/160725952-1a00d90d-022e-4df2-9a16-a2c94728b131.png">

Reply:
확실히 함수로 따로 빼니 가독성도 더 좋네요!
리뷰 감사합니다👍